src/AlertsAggregator.py

```diff
@@ -0,0 +1,222 @@
+                filtered_df = self.df[self.df['data'].isin(settlement_lst)]
+                filtered_df['time'] = pd.to_datetime(filtered_df['time']) 
+                # Filter the deteframe by the start and end time of the user
+                filtered_df = filtered_df[(filtered_df['time'].dt.hour >= start_hour) & (filtered_df['time'].dt.hour < end_hour)]
+                
+                # Create new column which represent the minutes by the nearest 15-minute interval
+                filtered_df['adjusted_time'] = filtered_df['time'].dt.strftime('%H:%M')
+                filtered_df['adjusted_time'] = pd.to_datetime(filtered_df['adjusted_time']).dt.round('15min').dt.strftime('%H:%M')
+                print(filtered_df)
+                
+                time_list = self.create_15_minute_interval_time_lst(start_time, end_time)
+                print(time_list)
+                adjusted_time_counts = filtered_df['adjusted_time'].value_counts().reindex(time_list, fill_value=0)
+                print(adjusted_time_counts)
+                return adjusted_time_counts
+            elif is_settlement_exist:
+                return {"message" : f"There were no alarms in the settlement: {settlement}"}
+            else:
+                raise ValueError("You selected a Settlement that does not exist.")
+        
+        except requests.exceptions.RequestException as req_err:
+            raise requests.exceptions.RequestException(f"Error occurred: {req_err}")
+        
+        except WrongSettlementException as settl_exc:
+            raise WrongSettlementException(settl_exc)
+
+        except ValueError as val_err:
+            raise ValueError(val_err)
+
+    # This function use create_quarter_hour_column func to return the best quarter of an hour to shower in
+    def best_time_to_shower(self, settlement : str, start_time : time, end_time : time):
+        try:    
+            adjusted_time_counts = self.create_adjusted_time_column(settlement, start_time, end_time)
+            print(f"{settlement} \n{adjusted_time_counts}")
+            if isinstance(adjusted_time_counts, pd.Series):
+                # Detect the quarter of an hour that appeared the less
+                best_time = adjusted_time_counts.idxmin()
+                return {"mesage" : f"The best time to take a shower is at the quarter: {best_time}"}
+            else:
+                return adjusted_time_counts
+        except ValueError as e:
+            raise ValueError(e)
+        except requests.exceptions.RequestException as req_err:
+            raise requests.exceptions.RequestException(f"Error occurred: {req_err}")
+   
+    
+    # This function use create_quarter_hour_column func to return the worst quarter of an hour to shower in
+    def worst_time_to_shower(self, settlement: str, start_time: time, end_time: time):
+        try:    
+            adjusted_time_counts = self.create_adjusted_time_column(settlement, start_time, end_time)
```

Comment:
add documentation inside the function to explain what you are doing here.
it is not clear at all.